src/partest/scala/tools/partest/nest/Runner.scala

```diff
@@ -447,7 +449,7 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
   /** Source files for the given test file. */
   def sources(file: File): List[File] = (
     if (file.isDirectory)
-      file.listFiles.toList filter (_.isJavaOrScala)
+      Files.walk(file.toPath).collect(Collectors.toList()).asScala.map(_.toFile).filter(_.isJavaOrScala).toList
```

Comment:
Preso tests put their (incomplete) mock user input into a subdirectory, so you can't do this for that. I support a special case for them, though, especially since I don't know if anyone still uses the presentation compiler. Does Metals?

Reply:
Also, many of the test mains in that directory look similar. If we were a cool test framework would it go only off of magic code comments?

Reply:
Thanks for the hint. I am awake now.